evm_loader/program/src/executor.rs

```diff
@@ -420,29 +423,24 @@ impl<'config, B: Backend> Machine<'config, B> {
                         caller: H160,
                         code: Vec<u8>,
                         transfer_value: U256,
-                        gas_limit: u64,
+                        _gas_limit: u64,
     ) -> ProgramResult {
-        debug_print!("create_begin gas_limit={}", gas_limit);
+        debug_print!("create_begin gas_limit={}", _gas_limit);
         let transaction_cost = gasometer::create_transaction_cost(&code);
         self.executor.state.metadata_mut().gasometer_mut()
             .record_transaction(transaction_cost)
             .map_err(|_| ProgramError::InvalidInstructionData)?;
 
-        let after_gas = self.executor.state.metadata().gasometer().gas();
-        let gas_limit = core::cmp::min(gas_limit, after_gas);
-
-        self.executor.state.metadata_mut().gasometer_mut().record_cost(gas_limit)
-            .map_err(|_| ProgramError::InvalidInstructionData)?;
-
         let scheme = evm::CreateScheme::Legacy { caller };
-        self.executor.state.enter(gas_limit, false);
 
         match self.executor.create(caller, scheme, transfer_value, code, None) {
```

Comment:
`gas_limit` should be used here instead of None

Reply:
Fixed